ocaml/xapi/xapi_network.ml

```diff
@@ -20,23 +20,26 @@ open D
 open Db_filter
 open Network
 
-let create_internal_bridge ~__context ~bridge ~uuid =
+let create_internal_bridge ~__context ~bridge ~uuid ~persist =
 	debug "Creating internal bridge %s (uuid:%s)" bridge uuid;
 	let dbg = Context.string_of_task __context in
 	let current = Net.Bridge.get_all dbg () in
 	if not(List.mem bridge current) then
 		let other_config = ["network-uuids", uuid] in
-		Net.Bridge.create dbg ~name:bridge ~other_config ()
+		Net.Bridge.create dbg ~name:bridge ~other_config ();
+	Net.Bridge.set_persistent dbg ~name:bridge ~value:persist
 
 let set_himn_ip ~__context bridge other_config =
 	let open Network_interface in
 	let dbg = Context.string_of_task __context in
 	try
 		let ip = List.assoc "ip_begin" other_config in
 		let netmask = List.assoc "netmask" other_config in
+		let persist = try List.assoc "persist" other_config |> bool_of_string with _ -> false in
 		let ipv4_conf =
 			(Static4 [Unix.inet_addr_of_string ip, netmask_to_prefixlen netmask]) in
 		Net.Interface.set_ipv4_conf dbg bridge ipv4_conf;
+		if persist then Net.Interface.set_persistent dbg bridge true;
```

Comment:
Perhaps just

    Net.Interface.set_persistent dbg bridge persist;

This way, you don't rely on networkd's default for the `persistent` field.